tests/cupyx_tests/scipy_tests/spatial_tests/test_distance.py

```diff
@@ -0,0 +1,123 @@
+import unittest
+
+import numpy
+import cupy
+try:
+    import scipy.spatial  # NOQA
+    import scipy.spatial.distance  # NOQA
+    scipy_available = True
+except ModuleNotFoundError:
+    scipy_available = False
+import cupyx.scipy.spatial.distance  # NOQA
+try:
+    import pylibraft  # NOQA
+    pylibraft_available = True
+except ModuleNotFoundError:
+    pylibraft_available = False
+from cupy import testing
+
+
+@testing.gpu
+@testing.with_requires("scipy")
+@testing.parameterize(*testing.product({
+    'dtype': ['float32', 'float64'],
+    'rows': [20, 100],
+    'cols': [20, 100],
+    'metric': ['euclidean', 'cityblock', 'canberra', 'chebyshev',
+               'hamming', 'correlation', 'jensenshannon', 'russellrao',
+               "minkowski", "cosine", "sqeuclidean"],
+    'p': [2.0],
+    'order': ["C", "F"]
+}))
+@unittest.skipUnless(scipy_available and pylibraft_available,
+                     'requires scipy and pylibcugraph')
+class TestCdist(unittest.TestCase):
```

Comment:
FYI generally CuPy is moving away from `unittest` in favor of `pytest`, so perhaps this should be changed to `@pytest.mark.skipif`.